nucypher-core-python/nucypher_core/__init__.pyi

```diff
@@ -429,22 +431,89 @@ class MetadataResponse:
         ...
 
 
+@final
+class AuthenticatedData:
+
+    def __init__(self, public_key: DkgPublicKey, conditions: Optional[Conditions]):
```

Comment:
@derekpierre I just recalled your comment on `Optional<&Condition>` in DKG structures. Are conditions supposed to be optional here too?

Reply:
It is downstream.

This is something that came to me last night, but it pervades many types in `nucypher-core` (ThresholdDecryptionRequest, ... etc.) and I didn't have time to fix it across all of them. I think we should fix it in a separate PR. Basically for DKG types Conditions should not be Optional but should be specified. Hence I filed #76 .